doc/source/dev/index.rst

```diff
@@ -1,6 +1,8 @@
-*********************
-SciPy Developer Guide
-*********************
+.. _core-dev-guide:
+
+==========================
+SciPy Core Developer Guide
+==========================
```

Comment:
As a new contributor, wouldn't you think you would want to _start_ with the "SciPy Developer Guide"? Most of this is really appropriate for new contributors, though. Some of it could be useful, so it's linked from the CTOC, but hopefully now this looks less like it's supposed to be starting point for new contributors.